modules/ldapbackend/powerldap.cc

```diff
@@ -195,17 +217,11 @@ bool PowerLDAP::getSearchEntry( int msgid, sentry_t& entry, bool dn, int timeout
         LDAPMessage* object;
 
 
-        if( ( i = waitResult( msgid, timeout, &result ) ) == LDAP_RES_SEARCH_RESULT )
-        {
-        	ldap_msgfree( result );
-        	return false;
-        }
+	if ( m_results.empty() )
+		return false;
 
-        if( i != LDAP_RES_SEARCH_ENTRY )
-        {
-        	ldap_msgfree( result );
-        	throw LDAPException( "Search returned an unexpected result" );
-        }
+	result = m_results.front();
+	m_results.pop_front();
```

Comment:
Again, `pop_front` is expensive. How about just keeping an index and doing cleanup at the end (or, as you do now, at the beginning of the next query)?